Server/ReasnAPI/ReasnAPI/Services/ImageService.cs

```diff
@@ -1,16 +1,145 @@
 ﻿using ReasnAPI.Models.Database;
+using ReasnAPI.Models.DTOs;
+using System.Linq.Expressions;
+using static System.Net.Mime.MediaTypeNames;
+using Image = ReasnAPI.Models.Database.Image;
+
+namespace ReasnAPI.Services;
+public class ImageService(ReasnContext context)
+{
+    public List<ImageDto> CreateImages(List<ImageDto> imageDtos)
+    {
+        var newImages = new List<Image>();
+
+        foreach (var imageDto in imageDtos)
+        {
+            var image = context.Images.FirstOrDefault(r => r.ObjectId == imageDto.ObjectId && r.ObjectTypeId == imageDto.ObjectTypeId);
+            if (image != null)
+            {
+                continue;
+            }
+
+            var newImage = new Image
+            {
+                ImageData = imageDto.ImageData,
+                ObjectId = imageDto.ObjectId,
+                ObjectTypeId = imageDto.ObjectTypeId
+            };
+
+            newImages.Add(newImage);
+        }
+
+        if (newImages.Any())
+        {
+            var objectType = context.ObjectTypes.FirstOrDefault(ot => ot.Id == newImages.First().ObjectTypeId);
+            if (objectType != null)
+            {
+                if (objectType.Name == "User" && newImages.Count > 1)
+                {
+                    context.Images.Add(newImages.First());
+                }
+                else if (objectType.Name == "Event")

```

Comment:
Extract "User" and "Event" strings into constant. Also - wouldn't be better to have some sort of enum on backend side for ObjectTypes, Statuses, Roles? Since it's hardcoded in DB anyways, so the ids won't change. This way we won't send another request to DB.